persistent/Database/Persist/Quasi.hs

```diff
@@ -197,18 +461,95 @@ removeSpaces =
     toLine' i = Line i . mapMaybe fromToken
 
     fromToken (Token t) = Just t
+    fromToken (DocComment t) = Just t
     fromToken Spaces{}  = Nothing
 
 -- | Divide lines into blocks and make entity definitions.
 parseLines :: PersistSettings -> [Line] -> [EntityDef]
 parseLines ps lines =
     fixForeignKeysAll $ toEnts lines
   where
-    toEnts (Line indent (name:entattribs) : rest) =
-        let (x, y) = span ((> indent) . lineIndent) rest
-        in  mkEntityDef ps name entattribs x : toEnts y
-    toEnts (Line _ []:rest) = toEnts rest
-    toEnts [] = []
+    toEnts :: [Line] -> [UnboundEntityDef]
+    toEnts =
+        map mk
+        . associateLines
+        . skipEmpty
```

Comment:
the old `toEnts` works just fine, but it forbids two adjacent lines to interact at all. And I need the doc comments preceeding a block to attach to the following block, so I had to rewrite.